framework/db/src/main/java/com/cloud/utils/crypt/EncryptionSecretKeyChanger.java

```diff
@@ -22,149 +22,332 @@
+        options.addOption(skipDatabaseMigration);
+        options.addOption(help);
+
+        return options;
+    }
+
+    private static HelpFormatter initializeHelper() {
+        HelpFormatter helper = new HelpFormatter();
+
+        helper.setOptionComparator((o1, o2) -> {
+            if (o1.isRequired() && !o2.isRequired()) {
+                return -1;
             }
-            // Old DB Key
-            if (arg.equals("-d")) {
-                oldDBKey = iter.next();
+            if (!o1.isRequired() && o2.isRequired()) {
+                return 1;
             }
-            // New MS Key
-            if (arg.equals("-n")) {
-                newMSKey = iter.next();
+            if (o1.hasArg() && !o2.hasArg()) {
+                return -1;
             }
-            // New DB Key
-            if (arg.equals("-e")) {
-                newDBKey = iter.next();
+            if (!o1.hasArg() && o2.hasArg()) {
+                return 1;
             }
-        }
+            return o1.getOpt().compareTo(o2.getOpt());
+        });
+
+        return helper;
+    }
+
+    private static boolean validateParameters(String oldMSKey, String oldDBKey, String newMSKey, String newDBKey,
+                                              String newEncryptorVersion, boolean loadNewMsKeyFromEnv) {
 
         if (oldMSKey == null || oldDBKey == null) {
-            System.out.println("Existing MS secret key or DB secret key is not provided");
-            usage();
-            return;
+            System.out.println("Existing Management secret key or DB secret key is not provided");
+            return false;
+        }
+
+        if (loadNewMsKeyFromEnv) {
+            if (StringUtils.isNotEmpty(newMSKey)) {
+                System.out.println("The new management key has already been set. Please check if it is set twice.");
+                return false;
+            }
+            newMSKey = System.getenv(ENV_NEW_MANAGEMENT_KEY);
+            if (StringUtils.isEmpty(newMSKey)) {
+                System.out.println("Environment variable " + ENV_NEW_MANAGEMENT_KEY + " is not set or empty");
+                return false;
+            }
         }
 
         if (newMSKey == null && newDBKey == null) {
-            System.out.println("New MS secret key and DB secret are both not provided");
-            usage();
-            return;
+            System.out.println("New Management secret key and DB secret are both not provided");
+            return false;
+        }
+
+        if (newEncryptorVersion != null) {
+            try {
+                CloudStackEncryptor.EncryptorVersion.fromString(newEncryptorVersion);
+            } catch (CloudRuntimeException ex) {
+                System.out.println(ex.getMessage());
+                return false;
+            }
         }
 
+        return true;
+    }
+
+    private static boolean migrateEverything(String oldMSKey, String oldDBKey, String newMSKey, String newDBKey,
```

Comment:
this is a god-method it could be renamed to `migrateAllEncryptedFields` and should be dissected to smaller methods.